src/yamlSettings.ts

```diff
@@ -0,0 +1,86 @@
+import { TextDocuments, Disposable, ClientCapabilities, WorkspaceFolder } from 'vscode-languageserver';
+import { CustomFormatterOptions, SchemaConfiguration } from './languageservice/yamlLanguageService';
+import { ISchemaAssociations } from './requestTypes';
+import { URI } from 'vscode-uri';
+import { JSONSchema } from './languageservice/jsonSchema';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+
+// Client settings interface to grab settings relevant for the language server
+export interface Settings {
+  yaml: {
+    format: CustomFormatterOptions;
+    schemas: JSONSchemaSettings[];
+    validate: boolean;
+    hover: boolean;
+    completion: boolean;
+    customTags: Array<string>;
+    schemaStore: {
+      enable: boolean;
+    };
+  };
+  http: {
+    proxy: string;
+    proxyStrictSSL: boolean;
+  };
+  editor: {
+    tabSize: number;
+  };
+}
+
+export interface JSONSchemaSettings {
+  fileMatch?: string[];
+  url?: string;
+  schema?: JSONSchema;
+}
+
+// This class is responsible for handling all the settings
+export class SettingsState {
+  public yamlConfigurationSettings: JSONSchemaSettings[] = undefined;
```

Comment:
It is matter of taste, but I think we can remove all `public` keyword, as all fields/method are public by default in TS